lib/sisimai/bite/email/google.rb

```diff
@@ -0,0 +1,320 @@
+        #
+        #   Technical details of temporary failure:
+        #   Google tried to deliver your message, but it was rejected by the recipient
+        #   domain. We recommend contacting the other email provider for further infor-
+        #   mation about the cause of this error. The error that the other server re-
+        #   turned was: 450 450 4.2.2 <mailboxfull@example.jp>... Mailbox Full (state 14).
+        #
+        #   -- OR --
+        #
+        #   Delivery to the following recipient failed permanently:
+        #
+        #        userunknown@example.jp
+        #
+        #   Technical details of permanent failure:=20
+        #   Google tried to deliver your message, but it was rejected by the server for=
+        #    the recipient domain example.jp by mx.example.jp. [192.0.2.59].
+        #
+        #   The error that the other server returned was:
+        #   550 5.1.1 <userunknown@example.jp>... User Unknown
+        #
+        return nil unless mhead['from']    =~ Re0[:from]
+        return nil unless mhead['subject'] =~ Re0[:subject]
+
+        require 'sisimai/address'
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        statecode0 = 0      # (Integer) The value of (state *) in the error message
+        v = nil
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+            next if e.empty?
+
+            # Technical details of permanent failure:=20
+            # Google tried to deliver your message, but it was rejected by the recipient =
+            # domain. We recommend contacting the other email provider for further inform=
+            # ation about the cause of this error. The error that the other server return=
+            # ed was: 554 554 5.7.0 Header error (state 18).
+            #
+            # -- OR --
```

Comment:
__[rubocop]__
* Style/NumericPredicate: Use `(readcursor & Indicators[:deliverystatus]).zero?` instead of `readcursor & Indicators[:deliverystatus] == 0`. [:ref](https://github.com/bbatsov/ruby-style-guide#predicate-methods)

<!-- Comment by [SideCI](https://sideci.com) -->